cvat-ui/src/components/quality-control/quality-control-page.tsx

```diff
@@ -166,89 +208,133 @@ function QualityControlPage(): JSX.Element {
         } finally {
+            setActiveTab(getTabFromHash(supportedTabs));
             dispatch(reducerActions.setFetching(false));
         }
     };
 
-    const onSaveQualitySettings = useCallback(async (values) => {
+    const onSaveQualitySettings = useCallback(async (settingsList: QualitySettings[], onError?: () => void) => {
+        const { settings, childrenSettings } = state.qualitySettings;
+
+        if (!settings) {
+            return;
+        }
+
         try {
-            const { settings } = state.qualitySettings;
-            if (settings) {
-                settings.targetMetric = values.targetMetric;
-                settings.targetMetricThreshold = values.targetMetricThreshold / 100;
-
-                settings.maxValidationsPerJob = values.maxValidationsPerJob;
-
-                settings.lowOverlapThreshold = values.lowOverlapThreshold / 100;
-                settings.iouThreshold = values.iouThreshold / 100;
-                settings.compareAttributes = values.compareAttributes;
-                settings.emptyIsAnnotated = values.emptyIsAnnotated;
-
-                settings.oksSigma = values.oksSigma / 100;
-                settings.pointSizeBase = values.pointSizeBase;
-
-                settings.lineThickness = values.lineThickness / 100;
-                settings.lineOrientationThreshold = values.lineOrientationThreshold / 100;
-                settings.orientedLines = values.orientedLines;
-
-                settings.compareGroups = values.compareGroups;
-                settings.groupMatchThreshold = values.groupMatchThreshold / 100;
-
-                settings.checkCoveredAnnotations = values.checkCoveredAnnotations;
-                settings.objectVisibilityThreshold = values.objectVisibilityThreshold / 100;
-
-                settings.panopticComparison = values.panopticComparison;
-                try {
-                    dispatch(reducerActions.setQualitySettingsFetching(true));
-                    const responseSettings = await settings.save();
-                    dispatch(reducerActions.setQualitySettings(responseSettings));
-                    notification.info({ message: 'Settings have been updated' });
-                } catch (error: unknown) {
-                    notification.error({
-                        message: 'Could not save quality settings',
-                        description: typeof Error === 'object' ? (error as object).toString() : '',
-                    });
-                    throw error;
-                } finally {
-                    dispatch(reducerActions.setQualitySettingsFetching(false));
-                }
+            dispatch(reducerActions.setQualitySettingsFetching(true));
+            const updatedSettings = await Promise.all(settingsList.map((setting) => setting.save()));
+            const updatedInstanceSettings = updatedSettings.find((setting) => setting.id === settings.id) || settings;
```

Comment:
Is it possible do not find own settings across all updated settings? 
If yes, let's use ?? operator.
If not, why do we need "default" value here?

Reply:
Yes, it is possible that there wont be `instance settings` in the provided array.
Used `??`.